kaldb/src/test/java/com/slack/kaldb/server/BulkIngestApiTest.java

```diff
@@ -208,49 +217,30 @@ public void testBulkApiBasic() throws Exception {
 
     // test with request1 twice. first one should succeed, second one will fail because of rate
     // limiter
-    CompletableFuture<AggregatedHttpResponse> response1 =
-        bulkApi
-            .addDocument(request1)
-            .aggregate()
-            .thenApply(
-                httpResponse -> {
-                  assertThat(httpResponse.status().isSuccess()).isEqualTo(true);
-                  assertThat(httpResponse.status().code()).isEqualTo(OK.code());
-                  BulkIngestResponse httpResponseObj = null;
-                  try {
-                    httpResponseObj =
-                        JsonUtil.read(httpResponse.contentUtf8(), BulkIngestResponse.class);
-                  } catch (IOException e) {
-                    fail("", e);
-                  }
-                  assertThat(httpResponseObj.totalDocs()).isEqualTo(1);
-                  assertThat(httpResponseObj.failedDocs()).isEqualTo(0);
-                  return httpResponse;
-                });
-
-    CompletableFuture<AggregatedHttpResponse> response2 =
-        bulkApi
-            .addDocument(request1)
-            .aggregate()
-            .thenApply(
-                httpResponse -> {
-                  assertThat(httpResponse.status().isSuccess()).isEqualTo(false);
-                  assertThat(httpResponse.status().code()).isEqualTo(TOO_MANY_REQUESTS.code());
-                  BulkIngestResponse httpResponseObj = null;
-                  try {
-                    httpResponseObj =
-                        JsonUtil.read(httpResponse.contentUtf8(), BulkIngestResponse.class);
-                  } catch (IOException e) {
-                    fail("", e);
-                  }
-                  assertThat(httpResponseObj.totalDocs()).isEqualTo(0);
-                  assertThat(httpResponseObj.failedDocs()).isEqualTo(0);
-                  assertThat(httpResponseObj.errorMsg()).isEqualTo("rate limit exceeded");
-                  return httpResponse;
-                });
-
-    await().until(response1::isDone);
-    await().until(response2::isDone);
+    AggregatedHttpResponse httpResponse = bulkApi.addDocument(request1).aggregate().join();
+    assertThat(httpResponse.status().isSuccess()).isEqualTo(true);
+    assertThat(httpResponse.status().code()).isEqualTo(OK.code());
+    try {
+      BulkIngestResponse httpResponseObj =
+          JsonUtil.read(httpResponse.contentUtf8(), BulkIngestResponse.class);
+      assertThat(httpResponseObj.totalDocs()).isEqualTo(1);
+      assertThat(httpResponseObj.failedDocs()).isEqualTo(0);
+    } catch (IOException e) {
+      fail("", e);
+    }
+
+    httpResponse = bulkApi.addDocument(request1).aggregate().join();
```

Comment:
this test was failing previously ( even without any of my changes ).

The reason this still worked was we don't implement `handle` and the AssertionError gets caught by the default `handle` and gets suppressed.